### PR TITLE
Correctly copy & paste data with context menu, and handle number notion that is based on locale

### LINKED ISSUE #282: Pasting from Context Menu trims numeric values
**Describe the bug**
When using the Paste function from the right-click Context Menu with default menu options (Cut, Copy, Paste) on numeric values, everything after the comma is truncated. 

**Current behavior**
Normal keyboard copy/cut/paste works fine. Context menu copy/cut works fine. But when using the right-click context menu to paste, it truncates the numeric values. Essentially anything after the comma if value is greater than 999.99 is truncated.

**Expected behavior**
Pasting numeric values from the right-click context menu should not truncate anything and paste the exact value copied.

**Screenshots or gifs**
copy_paste_bug

**Your environment details**
 - Device:  Desktop
 - OS: macOS
 - Browser Chrome/Chromium
 - React 18.2.0, Next.js 12.1.4

### PR DESCRIPTION
**Pull request by Michota**
## Context menu copy-pasting
- Fix issue #282 
- New function getRowsFromClipboard (asynchronous):
  - returns Compatible<Cell>[], which can be later used in pasteData function.
### Parse number with respecting locale number notion
Respect the locale settings for user-supplied numerical data, whether pasted or entered manually.
- New function parseLocaleNumber
  - Locale can be provided manually (as string). If this does not happen, the function will get it from the browser itself.
  - returns number.
```ts
function parseLocaleNumber(stringNumber: string, locale = getNavigatorLanguage()): number
```
```tsx
parseLocaleNumber("123456,78") // returns 123456.78
parseLocaleNumber("1 234 56,78", "pl-PL") // returns 123456.78
parseLocaleNumber("1,234,56.78", "en-US") // returns 123456.78

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -80,7 +80,6 @@
   },
   "husky": {
     "hooks": {
-      "pre-push": "if git-branch-is develop; then npx run-s eslint run:all:tests; fi"
```
Comment (DLowHP) [inline package.json]: I believe this was commited by mistake as I think we should keep this line.

File: src/lib/Functions/getNavigatorLanguage.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export const getNavigatorLanguage = (): string => navigator.languages && navigator.languages.length
+  ? navigator.languages[0]
+  :
+  // .userLanguage is a IE feature...
+  // navigator.userLanguage || 
+  navigator.language ||
+  // navigator.browserLanguage ||
+  "en-US";
```
Comment (DLowHP) [inline src/lib/Functions/getNavigatorLanguage.ts]: Does this need to be so complicated? I think that `navigator.language` should be sufficient, but I might be wrong?

File: tsconfig.json [hunk 3]
```diff
@@ -18,6 +18,7 @@
     "jsx": "react-jsx",
     "noImplicitAny": true,
     "noEmit": true,
+    "noEmitOnError": false,
```
Comment (DLowHP) [inline tsconfig.json]: Is this _explicitly_ needed? [The docs](https://www.typescriptlang.org/tsconfig#noEmitOnError:~:text=This%20defaults%20to%20false%2C%20making%20it%20easier%20to%20work%20with%20TypeScript%20in%20a%20watch%2Dlike%20environment%20where%20you%20may%20want%20to%20see%20results%20of%20changes%20to%20your%20code%20in%20another%20environment%20before%20making%20sure%20all%20errors%20are%20resolved.) say that this is `false` by default.

**Review by DLowHP — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (Michota) [reply package.json]: Yeah, You are right. But for some reason i can't push if this hook is there, even if i try to push it to non-develop branch...

File: src/lib/Functions/getNavigatorLanguage.ts (hunk 2 above)
Comment (Michota) [reply src/lib/Functions/getNavigatorLanguage.ts]: Oh, you are right, `navigator.language` is basically the same as `navigator.languages[0]`

File: package.json (hunk 1 above)
Comment (DLowHP) [reply package.json]: Will have to look into this some time... I'll add this to the backlog!

**Review by DLowHP — APPROVED** [review]

**Pull request merged**